src/core/StelLocationMgr.cpp

```diff
@@ -1158,16 +1158,28 @@ void StelLocationMgr::changeLocationFromNetworkLookup()
 
 			qDebug() << "Got location" << QString("%1, %2, %3 (%4, %5; %6)").arg(ipCity, ipRegion, ipCountry).arg(latitude).arg(longitude).arg(ipTimeZone) << "for IP" << locMap.value("ip").toString();
 
+			QString regionName = pickRegionFromCountryCode(ipCountryCode.toLower());
+			float luminance = StelLocation::DEFAULT_LIGHT_POLLUTION_LUMINANCE;
+			StelLocation ipLoc;
+			// Check location in our database and fetch light pollution luminance if it possible
+			if (!ipCity.isEmpty())
+				ipLoc = locationForString(QString("%1, %2").arg(ipCity, regionName));
+			else
+				ipLoc = pickLocationsNearby("Earth", longitude, latitude, 5.f).first();
```

Comment:
This will select a random location within 5° (~550km) of coordinates, which IMO does not make much sense. You would have to iterate over the map to find the entry closest to coordinates.